### PR TITLE
PLA-6616 add AutoML predictor, tests and docs

### PR DESCRIPTION
**Pull request by talltea**
# Citrine Python PR

## Description 
Adds automl support to citrine python!

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

**Comment by talltea** [comment]
I was able to create an AutoMLPredictor on development using this code: https://development.citrine-platform.com/projects/6a94ad36-5255-402a-851d-22366e750847/predictors/d52bf0c1-2e85-425f-b08b-36aad0cf3030/report although I am still debugging using that predictor in a PEW or DW

I was also able to use this code to locally test https://github.com/CitrineInformatics/nextgen-devkit/pull/834 (adding an automl predictor to hcep e2es)

@latture, I could use help looking over the docs and docstrings especially on this one.

File: src/citrine/informatics/predictors/auto_ml_predictor.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+from citrine.informatics.data_sources import DataSource
+from citrine.informatics.descriptors import Descriptor
+from citrine.informatics.reports import Report
+from citrine.informatics.predictors import Predictor
+
+__all__ = ['AutoMLPredictor']
+
+
+class AutoMLPredictor(Serializable['AutoMLPredictor'], Predictor):
+    """A predictor interface that builds a simgle model.
```
Comment (latture) [inline src/citrine/informatics/predictors/auto_ml_predictor.py:15]:
```suggestion
    """A predictor interface that builds a single ML model.
```

File: src/citrine/informatics/predictors/auto_ml_predictor.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    The model uses the set of inputs to predict the responses.
+    Only one value for responses is currently supported.
+    Supported complex inputs (such as chemical formulas) are auto-featurized.
```
Comment (latture) [inline src/citrine/informatics/predictors/auto_ml_predictor.py:19]: This isn't true. Users will need to add a formula featurizer if they want to featurize chemical formulas.

File: src/citrine/informatics/predictors/auto_ml_predictor.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    Only one machine learning model is built.
+
+    Parameters
+    ----------
+    name: str
+        name of the configuration
+    description: str
+        the description of the predictor
+    inputs: list[Descriptor]
+        Descriptors that represent inputs to relations
```
Comment (latture) [inline src/citrine/informatics/predictors/auto_ml_predictor.py:29]:
```suggestion
        Descriptors that represent inputs to the model
```

File: src/citrine/informatics/predictors/auto_ml_predictor.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    responses: list[Descriptor]
+        Descriptors that represent responses of relations
```
Comment (latture) [inline src/citrine/informatics/predictors/auto_ml_predictor.py:31]:
```suggestion
        Descriptors that represent responses of the model
```

**Comment by latture** [comment]
I made a few suggestions, but it looks great otherwise.

File: src/citrine/informatics/predictors/auto_ml_predictor.py (hunk 1 above)
Comment (talltea) [reply src/citrine/informatics/predictors/auto_ml_predictor.py:15]: ah yes, basic spelling is my nemesis XD

File: src/citrine/informatics/predictors/auto_ml_predictor.py (hunk 2 above)
Comment (talltea) [reply src/citrine/informatics/predictors/auto_ml_predictor.py:19]: I copy pasta-ed this from the simple ml predictor - is it true for that one, or should I also remove it there?

Comment (latture) [reply src/citrine/informatics/predictors/auto_ml_predictor.py:19]: It is true SimpleML, but not for this one.

Comment (latture) [reply src/citrine/informatics/predictors/auto_ml_predictor.py:19]: SimpleML does a little extra work to connect the graph and featurize common descriptors that AutoML does not.

**Review by latture — APPROVED** [review]

**Pull request merged**